core/domain/rte_component_registry.py

```diff
@@ -64,3 +67,64 @@ def get_tag_list_with_attrs(cls):
                 for ca_spec in component_specs['customization_arg_specs']]
 
         return component_tags
+
+    @classmethod
+    def get_component_types_to_component_classes(cls):
+        """Get component classes mapping for component types."""
+        module_fullname = 'extensions.rich_text_components.components'
+        module_obj = pkgutil.find_loader(module_fullname).load_module(
+            'components')
+        component_types_to_component_classes = {}
+        component_names = cls.get_all_rte_components().keys()
+        for component_name in component_names:
+            for name, obj in inspect.getmembers(module_obj):
+                if inspect.isclass(obj) and name == component_name:
+                    component_types_to_component_classes[
+                        'oppia-noninteractive-%s' % component_name.lower()] = (
+                            obj)
+
+        return component_types_to_component_classes
+
+    @classmethod
+    def get_inline_components(cls):
+        """Gets an list of inline component tags."""
+        rich_text_components_specs = cls.get_all_rte_components()
+        inline_component_tag_names = []
+        for component_spec in rich_text_components_specs.values():
+            if not component_spec['is_block_element']:
+                inline_component_tag_names.append(
+                    'oppia-noninteractive-%s' % component_spec['frontend_id'])
+        return inline_component_tag_names
+
+    @classmethod
+    def get_block_components(cls):
+        """Gets an list of block component tags."""
+        rich_text_components_specs = cls.get_all_rte_components()
+        block_component_tag_names = []
+        for component_spec in rich_text_components_specs.values():
+            if component_spec['is_block_element']:
+                block_component_tag_names.append(
+                    'oppia-noninteractive-%s' % component_spec['frontend_id'])
+        return block_component_tag_names
+
+    @classmethod
+    def get_simple_components(cls):
+        """Gets an list of simple component tags."""
+        rich_text_components_specs = cls.get_all_rte_components()
+        simple_component_tag_names = []
+        for component_spec in rich_text_components_specs.values():
+            if not component_spec['is_complex']:
```

Comment:
I'm glad to see this is in the RTE json spec. But then why have is_simple in the backend classes (in extensions/)?

Reply:
I had missed to remove it, fixed now.